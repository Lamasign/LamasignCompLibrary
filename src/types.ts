import {
	AccordionVariantsArray,
	AccordionVariantsEnum,
	AccordionVariantsMap,
} from "./components/accordion/BAccordion.types"


import {
	BadgeColorsArray,
	BadgeColorsEnum,
	BadgeColorsMap,
	BadgeSizeEnum,
	BadgeSizesArray,
	BadgeSizesMap,
	BadgeVariantsArray,
	BadgeVariantsEnum,
	BadgeVariantsMap,
} from './components/Badge/BBadge.types'

import {
	ButtonSizesEnum,
	ButtonVariantsEnum,
	ButtonTagsEnum,
	ButtonTypesEnum,
	ButtonSizesArray,
	ButtonSizesMap,
	ButtonVariantsArray,
	ButtonVariantsMap,
	ButtonTagsArray,
	ButtonTypesArray,
} from './components/Button/BButton.types'

import {
	CheckboxVariantsEnum,
	CheckboxVariantsArray,
	CheckboxsVariantsMap,
} from './components/Checkbox/BCheckbox.types'

import {
	DropdownVariantsEnum,
	DropdownDirectionEnum,
	DropdownDirectionMap,
	DropdownVariantsArray,
	DropdownVariantsMap,
} from './components/Dropdown/BDropdown.types'

import {
	FIELD_STATUS_INJECTION_KEY,
	FieldStatusesArray,
	FieldStatusesEnum,
	FieldStatusesClassesMap,
	FieldStatusesTypes,
	FieldMessageStatusesClassesMap
} from './components/Field/BField.types'

import {
	InputFileStatusesArray,
	InputFileStatusesClassesMap,
	InputFileStatusesEnum
} from './components/FileInput/BFileInput.types'

import {
	InputTypesEnum,
	InputTypesArray,
	InputStatusesArray,
	InputStatusesClassesMap,
	InputStatusesEnum,
	InputStatusesTypes,
	InputTypes,
} from './components/Input/BInput.types'

import {
	LinkDisabledClassesMap,
	LinkTags,
	LinkTagsArray,
	LinkTagsEnum,
	LinkVariantsArray,
	LinkVariantsEnum,
	LinkVariantsMap,
} from './components/Link/BLink.types'

import {
	MessageVariantsArray,
	MessageVariantsEnum,
	MessageVariantsMap,
} from './components/Message/BMessage.types'

import {
	NumberInputStatusesArray,
	NumberInputStatusesClassesMap,
	NumberInputStatusesEnum,
	NumberInputStatusesTypes,
} from './components/NumberInput/BNumberInput.types'

import {
	SelectColorsEnum,
	SelectColorsArray,
	SelectColorsMap,
} from './components/Select/BSelect.types'

import {
	SkeletonVariantsEnum,
	SkeletonVariantsArray,
	SkeletonVariantsMap,
	SkeletonAnimationsArray,
	SkeletonAnimationsEnum,
	SkeletonAnimationsMap,
	SkeletonColorsArray,
	SkeletonColorsEnum,
	SkeletonColorsMap,
} from './components/Skeleton/BSkeleton.types'

import {
	SwitchVariantsArray,
	SwitchVariantsEnum,
	SwitchVariantsMap,
} from './components/Switch/BSwitch.types'

import {
	TableColumnHeaderVariantsMap,
	TableColumnVariantsArray,
	TableColumnVariantsEnum,
	TableColumnVariantsMap,
} from './components/Table/BTable.types'

import {
	TagVariantsArray,
	TagVariantsEnum,
	TagVariantsMap,
	TagsVariantsMap,
} from './components/Tag/BTag.types'

import {
	ToastVariantsArray,
	ToastVariantsMap,
	ToastVariantsEnum,
} from './components/Toast/BToast.types'

import {
	TooltipPositionEnum,
	TooltipPositionsArray,
	TooltipPositionsMap,
	TooltipVariantsArray,
	TooltipVariantsEnum,
	TooltipVariantsMap,
} from './components/Tooltip/BTooltip.types'

export {
	AccordionVariantsArray,
	AccordionVariantsEnum,
	AccordionVariantsMap,
	BadgeColorsArray,
	BadgeColorsEnum,
	BadgeColorsMap,
	BadgeSizeEnum,
	BadgeSizesArray,
	BadgeSizesMap,
	BadgeVariantsArray,
	BadgeVariantsEnum,
	BadgeVariantsMap,
	ButtonSizesEnum,
	ButtonVariantsEnum,
	ButtonTagsEnum,
	ButtonTypesEnum,
	ButtonSizesArray,
	ButtonSizesMap,
	ButtonVariantsArray,
	ButtonVariantsMap,
	ButtonTagsArray,
	ButtonTypesArray,
	CheckboxVariantsEnum,
	CheckboxVariantsArray,
	CheckboxsVariantsMap,
	DropdownVariantsEnum,
	DropdownDirectionEnum,
	DropdownDirectionMap,
	DropdownVariantsArray,
	DropdownVariantsMap,
	FIELD_STATUS_INJECTION_KEY,
	FieldStatusesArray,
	FieldStatusesEnum,
	FieldStatusesClassesMap,
	FieldStatusesTypes,
	FieldMessageStatusesClassesMap,
	InputFileStatusesArray,
	InputFileStatusesClassesMap,
	InputFileStatusesEnum,
	InputTypesEnum,
	InputTypesArray,
	InputStatusesArray,
	InputStatusesClassesMap,
	InputStatusesEnum,
	InputStatusesTypes,
	InputTypes,
	LinkDisabledClassesMap,
	LinkTags,
	LinkTagsArray,
	LinkTagsEnum,
	LinkVariantsArray,
	LinkVariantsEnum,
	LinkVariantsMap,
	MessageVariantsArray,
	MessageVariantsEnum,
	MessageVariantsMap,
	NumberInputStatusesArray,
	NumberInputStatusesClassesMap,
	NumberInputStatusesEnum,
	NumberInputStatusesTypes,
	SelectColorsEnum,
	SelectColorsArray,
	SelectColorsMap,
	SkeletonVariantsEnum,
	SkeletonVariantsArray,
	SkeletonVariantsMap,
	SkeletonAnimationsArray,
	SkeletonAnimationsEnum,
	SkeletonAnimationsMap,
	SkeletonColorsArray,
	SkeletonColorsEnum,
	SkeletonColorsMap,
	SwitchVariantsArray,
	SwitchVariantsEnum,
	SwitchVariantsMap,
	TableColumnHeaderVariantsMap,
	TableColumnVariantsArray,
	TableColumnVariantsEnum,
	TableColumnVariantsMap,
	TagVariantsArray,
	TagVariantsEnum,
	TagVariantsMap,
	TagsVariantsMap,
	ToastVariantsArray,
	ToastVariantsMap,
	ToastVariantsEnum,
	TooltipPositionEnum,
	TooltipPositionsArray,
	TooltipPositionsMap,
	TooltipVariantsArray,
	TooltipVariantsEnum,
	TooltipVariantsMap,
}
