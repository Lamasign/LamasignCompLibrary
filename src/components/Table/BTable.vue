<template>
  <table class="flex shadow-lg bg-white">
    <slot></slot>
  </table>
</template>

<script lang="ts">
import { defineComponent } from '@vue/composition-api'

export default defineComponent({
  name: 'BTable',
})
</script>
