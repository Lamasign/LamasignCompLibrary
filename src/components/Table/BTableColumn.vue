<template>
  <div>
    <thead>
      <tr>
        <th class="bg-blue-100 border text-left px-8 py-4">{{ label }}</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td class="border px-8 py-4">
          <slot :item="item">
          </slot>
        </td>
      </tr>
    </tbody>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref } from '@vue/composition-api'

export default defineComponent({
  name: 'BTableColumn',
  props: {
    label: {
      type: String,
      required: false,
    },
    arrayData: {
      type: Array,
      default: () => [],
    },
  },
  setup(props) {
    const data = ref(props.arrayData)
    const item = {
      value: false,
      name: 'Ice cream sandwich',
      calories: 237,
      fat: 9.0,
      carbs: 37,
      protein: 4.3,
      iron: '1%'
    }
    return {
      data,
      item,
    }
  },
})
</script>
