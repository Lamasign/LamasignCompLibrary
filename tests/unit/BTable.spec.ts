import { createWrapper } from "./utils"
import BTable from '@/components/Table/BTable.vue'
import BTableColumn from '@/components/Table/BTableColumn.vue'
import { TableColumnVariantsArray } from "@/components/Table/BTable.types"

describe('Btable folder', () => {
	it('render comp Btable with correct classes', () => {
		const wrapper = createWrapper(BTable)
		expect(wrapper.html()).toContain('<table class="flex shadow-lg bg-white">')
	})

	it('render correct slot content on default slot', () => {
		const slotContent = 'TESTING'
		const wrapper = createWrapper(BTable, {
			slots: {
				default: slotContent
			}
		})
		expect(wrapper.text()).toMatch(slotContent)
	})

	it('render comp BTableColumn with correct classe', () => {
		expect.assertions(TableColumnVariantsArray.length + 1)
		const defaultWrapper = createWrapper(BTableColumn)

		expect(defaultWrapper.element).toMatchSnapshot()

		TableColumnVariantsArray.forEach(variant => {
			const wrapperWithProps = createWrapper(BTableColumn, {
				propsData: {
					variant
				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
		})
	})

	it('renders the correct <label> based on "label" prop', () => {
		const label = 'test de la prop label'
		const wrapper = createWrapper(BTableColumn, {
			propsData: {
				label
			}
		})
		expect(wrapper.html()).toContain(label)
		expect(wrapper.props().label).toEqual(label)
	})
})
