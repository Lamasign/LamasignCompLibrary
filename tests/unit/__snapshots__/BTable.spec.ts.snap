// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btable folder render comp BTableColumn with correct classe 1`] = `
<div>
  <thead>
    <tr>
      <th
        class="border text-center px-8 py-4 bg-blue-light text-blue-dark font-normal hover:bg-blue-dark focus:ring-4 ring-blue"
      >
        
      </th>
    </tr>
  </thead>
   
  <tbody>
    <tr
      class="flex flex-col"
    />
  </tbody>
</div>
`;

exports[`Btable folder render comp BTableColumn with correct classe 2`] = `
<div>
  <thead>
    <tr>
      <th
        class="border text-center px-8 py-4 bg-blue-light text-blue-dark font-normal hover:bg-blue-dark focus:ring-4 ring-blue"
      >
        
      </th>
    </tr>
  </thead>
   
  <tbody>
    <tr
      class="flex flex-col"
    />
  </tbody>
</div>
`;

exports[`Btable folder render comp BTableColumn with correct classe 3`] = `
<div>
  <thead>
    <tr>
      <th
        class="border text-center px-8 py-4 bg-orange-light text-blue-dark hover:bg-orange-dark focus:ring-4 ring-orange"
      >
        
      </th>
    </tr>
  </thead>
   
  <tbody>
    <tr
      class="flex flex-col"
    />
  </tbody>
</div>
`;

exports[`Btable folder render comp BTableColumn with correct classe 4`] = `
<div>
  <thead>
    <tr>
      <th
        class="border text-center px-8 py-4 bg-red-light text-blue-dark hover:bg-red-light focus:ring-4 ring-red"
      >
        
      </th>
    </tr>
  </thead>
   
  <tbody>
    <tr
      class="flex flex-col"
    />
  </tbody>
</div>
`;

exports[`Btable folder render comp BTableColumn with correct classe 5`] = `
<div>
  <thead>
    <tr>
      <th
        class="border text-center px-8 py-4 bg-green-light text-blue-dark hover:bg-green-dark focus:ring-4 ring-green"
      >
        
      </th>
    </tr>
  </thead>
   
  <tbody>
    <tr
      class="flex flex-col"
    />
  </tbody>
</div>
`;
